[pruner] add getter for pruned outputs in base pruner

kazhou opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63520
* #63519

Rather than having the user call `module.parametrizations.weight[0].pruned_outputs` each time they need to access the set of pruned indices, we add a getter `get_module_pruned_outputs` which takes the module as a argument and returns the set.

This will be used a lot when needing to call `manual_mask_update` on batch norm layers to match the pruned channels of the preceeding conv2d layer.

Differential Revision: [D30374558](https://our.internmc.facebook.com/intern/diff/D30374558/)

kazhou committed:
[pruner] add getter for pruned outputs in base pruner

Rather than having the user call `module.parametrizations.weight[0].pruned_outputs` each time they need to access the set of pruned indices, we add a getter `get_module_pruned_outputs` which takes the module as a argument and returns the set.

This will be used a lot when needing to call `manual_mask_update` on batch norm layers to match the pruned channels of the preceeding conv2d layer.

Differential Revision: [D30374558](https://our.internmc.facebook.com/intern/diff/D30374558/)

[ghstack-poisoned]

raghuramank100 commented on torch/ao/sparsity/experimental/pruner/base_pruner.py:
```diff
@@ -196,6 +196,11 @@ def convert(self):
+    def get_module_pruned_outputs(self, module):
+        r"""Returns the set of pruned indices of module"""
+        assert parametrize.is_parametrized(module)  # can only get pruned indices of pruned module
+        return module.parametrizations.weight[0].pruned_outputs
```
How do we know that this attr 'weight' is present in module.parametrizations?

kazhou replied:
I'll add a check that the module is in the pruner's `module_groups`. So far (for the sparsifier also), we assume that this is the only parametrization attached to `weight`.

kazhou committed:
Update on "[pruner] add getter for pruned outputs in base pruner"


Rather than having the user call `module.parametrizations.weight[0].pruned_outputs` each time they need to access the set of pruned indices, we add a getter `get_module_pruned_outputs` which takes the module as a argument and returns the set.

This will be used a lot when needing to call `manual_mask_update` on batch norm layers to match the pruned channels of the preceeding conv2d layer.

Differential Revision: [D30374558](https://our.internmc.facebook.com/intern/diff/D30374558/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add getter for pruned outputs in base pruner"


Rather than having the user call `module.parametrizations.weight[0].pruned_outputs` each time they need to access the set of pruned indices, we add a getter `get_module_pruned_outputs` which takes the module as a argument and returns the set.

This will be used a lot when needing to call `manual_mask_update` on batch norm layers to match the pruned channels of the preceeding conv2d layer.

Differential Revision: [D30374558](https://our.internmc.facebook.com/intern/diff/D30374558/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add getter for pruned outputs in base pruner"


Rather than having the user call `module.parametrizations.weight[0].pruned_outputs` each time they need to access the set of pruned indices, we add a getter `get_module_pruned_outputs` which takes the module as a argument and returns the set.

This will be used a lot when needing to call `manual_mask_update` on batch norm layers to match the pruned channels of the preceeding conv2d layer.

Differential Revision: [D30374558](https://our.internmc.facebook.com/intern/diff/D30374558/)

[ghstack-poisoned]
